internal/test/wmcb/wmcb_test.go

```diff
@@ -84,28 +90,37 @@ type pkgInfo struct {
 	shaType string
 }
 
-func initializeCniPluginsPkgInfo() error {
-	cniPluginsURL, err := framework.GetLatestCniPluginsURL()
-	if err != nil {
-		return fmt.Errorf("unable to get URL for CNI Plugins: %v", err)
+type wmcbFramework struct {
+	// TestFramework holds the instantiation of test suite being executed
+	*e2ef.TestFramework
+	// latestCniPluginsVersion is the latest 0.8.x version of CNI Plugins
+	LatestCniPluginsVersion string
+}
+
+// Setup initializes the wsuFramework.
+func (f *wmcbFramework) Setup(vmCount int, credentials []*types.Credentials, skipVMsetup bool) error {
+	f.TestFramework = &e2ef.TestFramework{}
+
+	if err := f.getLatestCniPluginsVersion(); err != nil {
+		return fmt.Errorf("unable to get latest 0.8.x version of CNI Plugins: %v", err)
+	}
+	if err := f.initializeCniPluginsPkgInfo(); err != nil {
+		return fmt.Errorf("unable to initialize CNI Plugins package info: %v", err)
 	}
-	cniPlugins.url = cniPluginsURL
 
-	cniPluginsSHA, err := framework.GetLatestCniPluginsSHA()
+	// Set up the framework
+	err := f.TestFramework.Setup(vmCount, credentials, skipVMsetup)
 	if err != nil {
-		return fmt.Errorf("unable to get SHA for CNI Plugins: %v", err)
+		fmt.Errorf("framework setup failed: %v", err)
```

Comment:
any specific reason for not returning here?

Reply:
Yeah you want to do `return fmt.Errorf("framework setup failed: %v", err)`

Reply:
Nice catch. Fixed it